colour/colorimetry/spectrum.py

```diff
@@ -53,6 +56,88 @@
            'zeros_spd',
            'ones_spd']
 
+DEFAULT_WAVELENGTH_DECIMALS = 10
+"""
+Default wavelength precision decimals.
+
+DEFAULT_WAVELENGTH_DECIMALS : int
+"""
+
+
+class SpectralMapping(ArbitraryPrecisionMapping):
+    """
+    Defines the base mapping for spectral data.
+
+    It enables usage of floating point wavelengths as keys by rounding them at
+    a specfic decimals count.
+
+    Parameters
+    ----------
+    data : dict or SpectralMapping, optional
+        Spectral data in a *dict* or *SpectralMapping* as follows:
+        {wavelength :math:`\lambda_{i}`: value,
+        wavelength :math:`\lambda_{i+1}`,
+        ...,
+        wavelength :math:`\lambda_{i+n}`}
```

Comment:
Should this be 
```
{wavelength :math:`\lambda_{i}`: value_0,
wavelength :math:`\lambda_{i+1}`: value_1,
...,
wavelength :math:`\lambda_{i+n}`: value_n}
```

I'm also not sure if `i` is a good variable name here, as it's supposed to denote a constant float (I assume), instead of a incrementing integer. Maybe just add a description of the domain of `i`.